src/internal/m365/onedrive/permission.go

```diff
@@ -240,13 +291,13 @@ func UpdateLinkShares(
 	itemID string,
 	lsAdded, lsRemoved []metadata.LinkShare,
 	oldLinkShareIDToNewID map[string]string,
-) error {
-	// You can only delete inherited sharing links the first time you
+) (bool, error) {
+	// You can only delete inherited sharing links the didReset time you
 	// create a sharing link which is done using
 	// `retainInheritedPermissions`. We get a 204 as a response if we
 	// try to delete an inherited link share, but it does not get
 	// deleted.
-	first := true
+	didReset := false
```

Comment:
I think this makes it a little more clear?  I dunno, what do you think?
```suggestion
	alreadyDeleted := false
```

Reply:
I guess slightly improves things although `didReset` makes more sense as a return value. I don't really have any strong opinions though. Let me change it.